jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/CrossContextDispatcher.java

```diff
@@ -251,7 +252,7 @@ public void forward(ServletRequest servletRequest, ServletResponse response) thr
         //if forwarding to the same environment we must mutate this request for Object wrapper identity
         if (_targetContext.getTargetContext().getContextHandler() instanceof ContextHandler.CoreContextHandler coreContextHandler)
         {
-            new Dispatcher(coreContextHandler.getContextHandler(), _uri, _decodedPathInContext).forward(httpServletRequest, httpServletResponse, DispatcherType.FORWARD);
+            new Dispatcher(coreContextHandler.getContextHandler(), _uri, URIUtil.encodePath(_decodedPathInContext)).forward(httpServletRequest, httpServletResponse, DispatcherType.FORWARD);
```

Comment:
Its a bit confusing with the changing back and forth between encoded, decoded or using the uri. We could change the signature of the constructor to remove the `decodedPathInContext` and pass in just the `context` and the `httpUri`. From that, we can derive all the info we need, avoiding re-encoding.

Reply:
Hmmm I think we should look at what is done in jetty-10/11 and mimic that as much as possible.  Will do....

Reply:
> Its a bit confusing with the changing back and forth between encoded, decoded or using the uri. We could change the signature of the constructor to remove the `decodedPathInContext` and pass in just the `context` and the `httpUri`. From that, we can derive all the info we need, avoiding re-encoding.

That's more or less what we do now in >=EE10